### PR TITLE
#160436728 Create channels for new engagements

### PR DESCRIPTION
**Pull request by nyandiekaFelix**
#### What does this PR do?
Enables the creation of new slack channels for a new partner engagement.

#### Description of Task to be completed?
When a developer is placed, `p-partner` and `p-partner-int` slack channels should be created. This happens in the event it's a new partner.

#### How should this be manually tested?

Assuming you are an administrator of the workspace:
- Create an [app](https://api.slack.com/apps). It should have workspace privileges in order to create channels(`groups:write`) 
- Generate a token for the app and add it to the `.env` file as the `SLACK_TOKEN`
- Calling the `createPartnerChannels` function in `server/modules/slackIntegration.js`, while providing the partner's details, will create a **private** channel for a specific partner engagement

#### What are the relevant pivotal tracker stories?
[160436728](https://www.pivotaltracker.com/story/show/160436728)

### CODE REVIEW

**Comment by Kelechukwu** [comment]
@nyandiekaFelix please share your slack api token via slack

**Comment by psibomana** [comment]
[Maintainability](https://codeclimate.com/github/andela/bp-esa-backend/maintainability)

File: server/modules/slackIntegration.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ * @desc Create slack channels for a partner engagement
+ *
+ * @param {string} partnerId - ID of the partner
+ * @returns {Object} - An object containing details of the created channels
+ */
+export const createPartnerChannels = async (partnerId) => {
+  try {
+    let partner;
+
+    if (process.env.NODE_ENV === 'test') {
```
Comment (Kelechukwu) [inline server/modules/slackIntegration.js]: This should be controlled from the test case and not written into the body of the function

File: server/modules/slackIntegration.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      throw new Error('Partner record was not found');
+    }
+
+    const channelNames = makeChannelNames(partner.name);
+    const { generalChannel, internalChannel } = channelNames;
+
+    let createGeneral;
+    let createInternal;
+
+    if (process.env.NODE_ENV === 'test') {
```
Comment (Kelechukwu) [inline server/modules/slackIntegration.js]: This should be controlled from the test case and not written into the body of the function

**Comment by Kelechukwu** [comment]
@nyandiekaFelix please look at the comments i made on lines 42 and 50 about having `if (process.env.NODE_ENV === 'test')` in the body of the function .

**Comment by nzediegwu1** [comment]
This PR has been combined into [#160436611 Slack channel services](https://github.com/andela/bp-esa-backend/pull/34), and therefore is no longer needed

**Pull request closed without merging**